File: backend/src/test/java/lan/dk/podcastserver/utils/facade/stats/StatsPodcastTypeTest.java
package lan.dk.podcastserver.utils.facade.stats;

import io.vavr.collection.HashSet;
import io.vavr.collection.Set;
import com.github.davinkevin.podcastserver.business.stats.NumberOfItemByDateWrapper;
import lan.dk.podcastserver.business.stats.StatsPodcastType;
import org.junit.Test;

import java.time.LocalDate;
import java.time.Month;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Created by kevin on 01/07/15 for Podcast Server
 */
public class StatsPodcastTypeTest {


    private static final NumberOfItemByDateWrapper NI_1 = new NumberOfItemByDateWrapper(LocalDate.of(2015, Month.JULY, 1), 100);
    private static final NumberOfItemByDateWrapper NI_2 = new NumberOfItemByDateWrapper(LocalDate.of(2015, Month.JULY, 2), 200);
    private static final NumberOfItemByDateWrapper NI_3 = new NumberOfItemByDateWrapper(LocalDate.of(2015, Month.JULY, 3), 300);
    private static final String FAKE_TYPE = "FakeType";

    @Test
    public void should_has_correct_value() {
        /* Given */
        Set<NumberOfItemByDateWrapper> numberOfItemSets = HashSet.of(NI_1, NI_2, NI_3);

        StatsPodcastType statsPodcastType = new StatsPodcastType(FAKE_TYPE, numberOfItemSets);

        /* When */
        String type = statsPodcastType.getType();
        Set<NumberOfItemByDateWrapper> values = statsPodcastType.getValues();

        /* Then */
        assertThat(type).isEqualTo(FAKE_TYPE);
        assertThat(values).contains(NI_1, NI_2, NI_3);
    }
}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/business/stats/StatsPodcastTypeTest.kt
package com.github.davinkevin.podcastserver.business.stats

import io.vavr.API.Set
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDate
import java.time.Month

/**
 * Created by kevin on 01/07/15 for Podcast Server
 */
class StatsPodcastTypeTest {

    @Test
    fun `should has correct value`() {
        /* Given */
        val numberOfItemSets = Set(NI_1, NI_2, NI_3)

        /* When */
        val (type, values) = StatsPodcastType(FAKE_TYPE, numberOfItemSets)

        /* Then */
        assertThat(type).isEqualTo(FAKE_TYPE)
        assertThat(values).contains(NI_1, NI_2, NI_3)
    }

    companion object {
        private val NI_1 = NumberOfItemByDateWrapper(LocalDate.of(2015, Month.JULY, 1), 100)
        private val NI_2 = NumberOfItemByDateWrapper(LocalDate.of(2015, Month.JULY, 2), 200)
        private val NI_3 = NumberOfItemByDateWrapper(LocalDate.of(2015, Month.JULY, 3), 300)
        private const val FAKE_TYPE = "FakeType"
    }
}
